frontend/apps/reader/modules/readerpagemap.lua

```diff
@@ -39,20 +42,47 @@ function ReaderPageMap:init()
     self.ui:registerPostInitCallback(function()
         self:_postInit()
     end)
+    self.ui.menu:registerToMainMenu(self)
 end
 
 function ReaderPageMap:_postInit()
     self.initialized = true
-    if self.ui.document.info.has_pages then
-        return
+    -- chars_per_synthetic_page is saved to cr3 cache by crengine on first building of synthetic pagemap
+    -- after that synthetic pagemap is built on doc rendering without asking doc settings
+    -- different combinations of doc settings and cached chars_per_synthetic_page must be handled
```

Comment:
Period at end of first line, uppercase at start of second line (I think, otherwise I would read it as a single sentence and it would need a comma, and may make no sense.)
Uppercase also on the 3rd line.
But I still don't understand the 2nd sentence: "After that, synthetic pagemap is built on doc rendering without asking doc settings" ?

May be just:
"It's possible that the crengine doc cache is inconsistent with our setting (cache from a past opening, settings sync'ed from another device. Make sure we're consistent, honoring the cre cache values to avoid a document reload." ?